tamr_unify_client/dataset/resource.py

```diff
@@ -434,3 +444,88 @@ def _geo_attr_names():
             "polygon",
             "multiPolygon",
         }
+
+
+class DatasetSpec:
+    """A representation of the server view of a dataset."""
+
+    def __init__(self, client, data, api_path):
+        self._data = data
+        self.client = client
+        self.api_path = api_path
+
+    @staticmethod
+    def of(dataset):
+        """Creates a dataset spec from a dataset.
+
+        :param dataset: The existing dataset.
+        :type dataset: :class:`~tamr_unify_client.dataset.resource.Dataset`
+        :return: The corresponding dataset spec.
+        :rtype: :class:`~tamr_unify_client.dataset.resource.DatasetSpec`
+        """
+        return DatasetSpec(dataset.client, deepcopy(dataset._data), dataset.api_path)
+
+    def from_data(self, data):
+        """Creates a spec with the same client and API path as this one, but new data.
+
+        :param data: The data for the new spec.
+        :type data: dict
+        :return: The new spec.
+        :rtype: :class:`~tamr_unify_client.dataset.resource.DatasetSpec`
+        """
+        return DatasetSpec(self.client, data, self.api_path)
+
+    def to_dict(self):
+        """Returns a copy of this spec's internal representation.
```

Comment:
The contract here isn't really 'this spec's internal representation' - rather, the contract should be that `to_dict()` will return a `dict` that conforms to the API of `PUT datasets/{id}`